pkg/api/ui_handler.go

```diff
@@ -80,6 +80,8 @@ func NewHandlerWithDefault(root http.FileSystem, handler http.Handler, defaultPa
 		}
 		_, err := root.Open(path.Clean(urlPath))
 		if err != nil && os.IsNotExist(err) {
+			fmt.Println("redirecting...")
+			http.Redirect(w, r, defaultPath, http.StatusFound)
```

Comment:
As far as I remember, you need to return after redirect and not continue the code flow